api/v1alpha3/awsmachine_webhook_test.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	"testing"
+
+	"k8s.io/utils/pointer"
+)
+
+func TestAWSMachine_ValidateUpdate(t *testing.T) {
+	tests := []struct {
+		name    string
+		old     *AWSMachine
+		new     *AWSMachine
```

Comment:
nit: I usually try to avoid using `new` as a variable name, given that it's a go keyword. If you want to change the name, you could use something like `updated` instead.